### PR TITLE
feat: add cms component schematic

### LINKED ISSUE #12: Create custom component schematic
We can provide an angular schematic to simplify the creation of custom components, routes, page templates, etc. This will help to:
- customise very fast
- customise according to our best practice
- avoid reading documentation

Angular schematics are the technique that can be used for this.

### PR DESCRIPTION
**Pull request by znikola**
closes #12

### CODE REVIEW

File: projects/schematics/src/add-cms-component/index.ts [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+      const buffer = tree.read(templatePath);
+      templateContent = buffer ? buffer.toString(UTF_8) : '';
+      startIndex = templateContent.length;
+    }
+
+    if (Boolean(templateContent)) {
+      const insertion =
+        `  <ng-container *ngIf="${CMS_COMPONENT_DATA_PROPERTY_NAME}$ | async as data">` +
+        `\n      {{data | json}}` +
+        `\n    </ng-container>`;
```
Comment (psmul) [inline projects/schematics/src/add-cms-component/index.ts]: When component is created, result html has weird intendation.

File: projects/schematics/src/add-cms-component/schema.json [hunk 2]
```diff
@@ -0,0 +1,160 @@
+{
+  "$schema": "http://json-schema.org/schema",
+  "id": "SpartacusSchematics",
+  "title": "Spartacus Schematics",
+  "type": "object",
+  "properties": {
+    "declaringCmsModule": {
```
Comment (psmul) [inline projects/schematics/src/add-cms-component/schema.json]: declaringCmsModule sounds/reads weird to me, maybe just `declareCmsModule`?

File: projects/schematics/src/add-cms-component/index.ts (hunk 1 above)
Comment (znikola) [reply projects/schematics/src/add-cms-component/index.ts]: Is there an API that can return us the existing indentation level?

**Comment by znikola** [comment]
>     1. Lack of created cms component import in imports ConfigModule configuration

Good catch, this is caused when `skipImport=true`. 
 
>     2. Maybe OnInit is unnecessary when we create a new component?

That's how Angular generates the component by default, and there's no an option to easily switching it off. Do you think it's worth the effort to remove it after generating?

>     3. When adding created cms component to app.module.ts, a glitch with 2 commas in imports appears

I can't reproduce this with `ng g .:add-cms-component myAwesomeCms --cms-model=MyModel --declareCmsModule=app`. Can you explain it to me how to reproduce it?

File: projects/schematics/src/add-cms-component/index.ts (hunk 1 above)
Comment (znikola) [reply projects/schematics/src/add-cms-component/index.ts]: Scratch that, I decided to put everything in one line

**Comment by znikola** [comment]
Responding:

> Additional things found after next iteration of QA:
> 
>     1. When used:
>        `:add-cms-component myAwesomeCms --cms-model=MyModel`
>        `my-awesome-cms.module.ts` has wrong component import:
>        `import { MyAwesomeCmsComponent } from './my-awesome-cms.component.ts';`
>        Should not contain '.ts' extension.

Thanks, good catch. Addressed.

>     2. When used:
>        `:add-cms-component myAwesomeCms --cms-model=MyModel`
>        `my-awesome-cms.module.ts` has doubled component import in `imports` (ConfigModule and main imports array)

Again, great catch, addressed.

>     3. I believe we should follow Spartacus module style. If we import component in module file, we should also include declarations like **declarations, entryComponents, exports**. Although if we specify `--module=app` it works fine (adds component to entryComponents and exports).

Good point, implemented.

>     4. When used:
>        `ng g @spartacus/schematics:add-cms-component myAwesomeCms --cms-model=MyModel --module=app --declareCmsModule=app`
>        (I know it's falsy scenario). It also adds module with '.ts' extension.
>        `import { App } from './app.module.ts';`

Addressed with the 1st point.

**Pull request merged**